real-main/app/handlers/appsync/handlers.py

```diff
@@ -1416,19 +1416,18 @@ def lambda_server_error(caller_user_id, arguments, context=None, **kwargs):
     raise Exception(f'Test of lambda server error, request `{request_id}`')
 
 
-@routes.register('Query.findUsers')
+@routes.register('Query.findContacts')
 @validate_caller
 @update_last_client
-def find_users(caller_user, arguments, **kwargs):
-    emails = arguments['emails'] or []
-    phones = arguments['phoneNumbers'] or []
-
-    if not emails and not phones:
-        raise ClientException('Called without any arguments... probably not what you intended?')
-
-    if len(emails) + len(phones) > 100:
-        raise ClientException('Cannot submit more than 100 combined emails and phoneNumbers')
-
-    user_ids = user_manager.find_users(caller_user, emails=emails, phones=phones)
-    caller_user.update_last_found_users_at(now=pendulum.now('utc'))
-    return {'items': user_ids}
+def find_contacts(caller_user, arguments, **kwargs):
+    contacts = arguments['contacts']
+
+    if len(contacts) > 100:
+        raise ClientException('Cannot submit more than 100 contact inputs')
+
+    contact_id_to_user_id = user_manager.find_contacts(caller_user, contacts=contacts)
+    caller_user.update_last_found_contacts_at(now=pendulum.now('utc'))
```

Comment:
Even though I don't know if these actually do throw a `UserException`, I think it's still better to wrap them in this try-catch block because they may be changed in the future to do so.

Also, `contacts` is no longer a kwarg (it's just a normal positional argument) so I wouldn't use the `contacts=contacts` syntax on it, I would just pass in `contacts`.

```suggestion
    try:
        contact_id_to_user_id = user_manager.find_contacts(caller_user, contacts)
        caller_user.update_last_found_contacts_at(now=pendulum.now('utc'))
    except UserException as err:
        raise ClientException(str(err)) from err
        
```